uxarray/grid.py

```diff
@@ -228,6 +258,118 @@ def __init_ds_var_names__(self):
+                    setattr(self, key, self.ds[value])
+
+            # Present Coordinate Names
+            if self.ds.coords is not None:
+                if value in self.ds.coords:
+                    setattr(self, key, self.ds[value])
+
+    def integrate(self, var_key, quadrature_rule="triangular", order=4):
+        """ Integrates over all the faces of the given mesh.
+        Parameters
+        ----------
+
+        var_key : string, required
+            Name of variable for integration.
+
+        Returns
+        -------
+
+        double: integration result.
+
+        Examples
+        --------
+
+        Open grid file only
+        >>> grid = ux.open_dataset("grid.ug", "centroid_pressure_data_ug")
+
+
+        Open grid file along with data
+        >>> integral_psi = grid.integrate("psi")
+        """
+        integral = 0.0
+
+        # call function to get area of all the faces as a np array
+        face_areas = self.get_face_areas(quadrature_rule, order)
+
+        face_vals = self.ds.get(var_key).to_numpy()
+        integral = np.dot(face_areas, face_vals)
+
+        return integral
+
+    def get_face_areas(self, quadrature_rule="triangular", order=4):
+        """Face areas calculation function for grid class, calculates area of
+        all faces in the grid.
+
+        Returns
+        -------
+
+        area of all the faces in the mesh. : ndarray
+
+        Examples
+        --------
+
+        Open a uxarray grid file
+
+        >>> grid = ux.open_dataset("/home/jain/uxarray/test/meshfiles/outCSne30.ug")
+
+        Get area of all faces in the same order as listed in grid.ds.Mesh2_face_nodes
+
+        >>> grid.get_face_areas
+        array([0.00211174, 0.00211221, 0.00210723, ..., 0.00210723, 0.00211221,
+            0.00211174])
+        """
+        if self._face_areas is None:
+            # area of a face call needs the units for coordinate conversion if spherical grid is used
+            coords_type = "spherical"
+            if not "degree" in self.Mesh2_node_x.units:
+                coords_type = "cartesian"
+
+            face_nodes = self.Mesh2_face_nodes.data
+            dim = self.Mesh2.attrs['topology_dimension']
+
+            # initialize z
+            z = np.zeros((self.ds.nMesh2_node.size))
+
+            # call func to cal face area of all nodes
+            x = self.Mesh2_node_x.data
+            y = self.Mesh2_node_y.data
+            # check if z dimension
+            if self.Mesh2.topology_dimension > 2:
+                z = self.Mesh2_node_z.data
+
+            # call function to get area of all the faces as a np array
+            self._face_areas = get_all_face_area_from_coords(
+                x, y, z, face_nodes, dim, quadrature_rule, order, coords_type)
+
+        return self._face_areas
+
+    # use the property keyword for declaration on face_areas property
+    face_areas = property(get_face_areas)
+
     def __init_grid_var_attrs__(self):
```

Comment:
It looks like there are two definitions for this function

Reply:
Good catch! Fixing this in PR #74